src/controllers/DynamicFacetQueryController.ts

```diff
@@ -6,13 +6,24 @@ import { QueryEvents } from '../events/QueryEvents';
 import { findIndex } from 'underscore';
 import { IQueryResults } from '../rest/QueryResults';
 import { IDynamicFacet } from '../ui/DynamicFacet/IDynamicFacet';
+import { DynamicFacetRequestBuilder } from './DynamicFacetRequestBuilder';
+import { IQuery } from '../rest/Query';
 
 export class DynamicFacetQueryController {
+  protected requestBuilder: DynamicFacetRequestBuilder;
   private numberOfValuesToRequest: number;
   private freezeCurrentValues = false;
   private freezeFacetOrder = false;
 
   constructor(protected facet: IDynamicFacet) {
+    this.requestBuilder = new DynamicFacetRequestBuilder({
+      facetId: this.facet.options.id,
+      field: this.facet.fieldName,
+      type: this.facet.facetType,
+      sortCriteria: this.facet.options.sortCriteria as FacetSortCriteria,
```

Comment:
Rather than casting, should the `sortCriteria` option typing be adjusted?

Reply:
I wanted to use type guards at first but it doesn't validate in TSv1. But I can typecast is directly in the post processing of the option instead